spring-security-modules/spring-security-web-boot-4/src/test/java/com/baeldung/apikeysecretauth/controller/ProductControllerUnitTest.java

```diff
@@ -0,0 +1,96 @@
+package com.baeldung.apikeysecretauth.controller;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.baeldung.apikeysecretauth.repository.UserKeysRepository;
+import com.baeldung.apikeysecretauth.repository.model.UserData;
+import com.baeldung.apikeysecretauth.repository.model.UserKeysData;
+import com.baeldung.apikeysecretauth.service.ProductService;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(ProductController.class)
+public class ProductControllerUnitTest {
```

Comment:
JUnit 5 doesn't need classes or methods to be `public`

Reply:
fixed